include/envoy/thread_local/thread_local.h

```diff
@@ -228,5 +229,24 @@ class Instance : public SlotAllocator {
   virtual Event::Dispatcher& dispatcher() PURE;
 };
 
+struct MainThread {
+  using MainThreadSingleton = InjectableSingleton<MainThread>;
+  bool inMainThread() { return main_thread_id_ == std::this_thread::get_id(); }
+  static void init() {
+    static MainThread main_thread;
```

Comment:
Is there a need to have something `static` outside of the `InjectableSingleton`?

Reply:
Because in the current implementation, the raw pointer passed to InjectableSingleton doesn't own the object. I will change the implementation and free the MainThread object manually.